src/Andre/SoulsFormats/SoulsFormats/Formats/DCX.cs

```diff
@@ -427,12 +475,42 @@ internal static void Compress(Span<byte> data, BinaryWriterEx bw, Type type)
                 CompressDCXKRAK(data, bw);
             else if (type == Type.DCX_KRAK_MAX)
                 CompressDCXKRAK(data, bw, 9);
+            else if (type == Type.ZSTD)
+            {
+                // Temporary until proper ZSTD re-compression settings are discovered
+                CompressDCPDFLT(data, bw);
+            }
             else if (type == Type.Unknown)
                 throw new ArgumentException("You cannot compress a DCX with an unknown type.");
             else
                 throw new NotImplementedException("Compression for the given type is not implemented.");
         }
 
+        private static void CompressDCXZSTD(Span<byte> data, BinaryWriterEx bw)
+        {
+            // TODO: just a stub for future implementation
+            /*
+            bw.WriteASCII("DCX\0");
+            bw.WriteASCII("ZSTD");
+            bw.WriteInt32(0x20);
+            bw.WriteInt32(0x9000000);
+            bw.WriteInt32(0);
+            bw.WriteInt32(0);
+            bw.WriteInt32(0);
+            bw.WriteInt32(0x00010100);
+
+            bw.WriteASCII("DCS\0");
+            bw.WriteInt32(data.Length);
+            bw.ReserveInt32("CompressedSize");
+
+            int compressedSize = SFUtil.WriteZstd(bw, 3, data);
+            bw.FillInt32("CompressedSize", compressedSize);
+
+            bw.WriteASCII("DCA\0");
+            bw.WriteInt32(8);
+            */
+        }
+
```

Comment:
I'm still hopeful we'll figure this out, though it seems more and more likely we'll need to inject some hardcore ghidra user like chain with a few shots of caffeine